synfig-studio/src/gui/widgets/widget_timetrack.cpp

```diff
@@ -198,6 +198,47 @@ bool Widget_Timetrack::move_selected(synfig::Time delta_time)
 	return ok;
 }
 
+ void Widget_Timetrack::interpolate_selected(synfig::Interpolation type)
+{
+	synfig::Waypoint::Model model;
+	model.set_before(type);
+	model.set_after(type);
+	std::vector<WaypointItem*> selection = waypoint_sd.get_selected_items(); //storing selected waypoint items
+	std::vector<std::set<synfig::Waypoint, std::less<synfig::UniqueID> >> accumilated_selection_set;
+	accumilated_selection_set.clear();
+
+    for(int i =0 ; i<selection.size() ; i++) //iteratre through selection size
+    {
+		std::set<synfig::Waypoint, std::less<synfig::UniqueID> > waypoint_set_new; //declaration
+		fetch_waypoints(*selection[i], waypoint_set_new); //setting up waypoint_set_new
+		std::set<synfig::Waypoint, std::less<synfig::UniqueID> >::const_iterator iter;
+		accumilated_selection_set.push_back(waypoint_set_new);
+
+		for(iter=accumilated_selection_set[i].begin();iter!=accumilated_selection_set[i].end();++iter)
+        {
+			synfigapp::Action::PassiveGrouper group(get_canvas_interface()->get_instance().get(),_("Change Waypoint Group"));
+
```

Comment:
It seems you'll:
1. create multiples Change Waypoint Group groups
2. set the waypoints from previous iterations multiple times

Reply:
how about now?

p.s. im uploading the commit in 5 mins if you see this rn